packages/scaffold-config/src/frameworks.ts

```diff
@@ -14,6 +14,43 @@ export type WizardBundler = typeof dependencies.WIZARD_BUNDLERS[number]
 
 export type CodeGenFramework = Cypress.ResolvedComponentFrameworkDefinition['codeGenFramework']
 
+export async function isDependencyInstalledByName (packageName: string, projectPath: string): Promise<{dependency: string, detectedVersion: string | null}> {
+  try {
+    debug('detecting %s in %s', packageName, projectPath)
+
+    const packageFilePath = resolvePackagePath(packageName, projectPath, false)
+
+    if (!packageFilePath) {
+      debug('unable to resolve dependency %s', packageName)
+
+      return {
+        dependency: packageName,
+        detectedVersion: null,
+      }
```

Comment:
Feels kinda awkward having three different returns statements here - might be a bit cleaner to do something like

```ts
let detectedVersion: string | null;
try {
  ...
} catch () {
  ...
}

return {
  dependency: packageName,
  detectedVersion
}
```

Reply:
Refactored this to make it more clean